DT-457: Add ability to select all stories in a project

torresga opened the pull request:
### Jira Ticket 
See https://ombulabs.atlassian.net/browse/DT-457

### Motivation / Context

In a Points project, stories need to be selected individually. There is no option to select all. There should be a Select All option to be able to easily perform bulk actions.

### QA / Testing Instructions
1. Go to a project with stories
2. Check the Select All checkbox
3. Ensure that all stories are selected and Bulk Delete button becomes enabled and shows the number of all stories selected
4. Unselect the Select All checkbox
5. Ensure that no stories are selected and the Bulk Delete button is disabled


### Screenshots:
<img width="1236" alt="Screenshot 2024-10-21 at 10 07 19 AM" src="https://github.com/user-attachments/assets/866a8f33-e609-4cfe-b100-c1bb78062777">
<img width="1236" alt="Screenshot 2024-10-21 at 10 07 12 AM" src="https://github.com/user-attachments/assets/a2d24fdf-552e-42e3-b259-a37b1f7d7e40">

Note:

* `app/models/user.rb` was changed here because there were some linting issues standardrb wouldn’t let me commit without. 
* I also realize the CI for this is failing - I did add a PR for this change in another PR https://github.com/fastruby/points/pull/332 and then cherry-picked the change in this PR to ensure CI passed.

___

I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/CODE_OF_CONDUCT.md).

torresga committed:
Adds Select All checkbox and functionality

torresga committed:
Adds indeterminate state if not all checkboxes are selected

torresga committed:
Adds specs to test Select All checkbox

JuanVqz commented on app/assets/javascripts/project.js:
```diff
@@ -118,3 +114,40 @@ function toggleCloneSubProjects(value) {
+    }
+}
+
+function updateSelectAllStatus() {
```
I have a question, @torresga you don't need to check for the `unlocked` when updating the states of these check boxes?

JuanVqz committed:
Gate Select All on unlocked projects and harden select-all JS

- Only render the Select All checkbox when the project is unlocked, matching
  Add a Story and the bulk-delete control (addresses the review question about
  the unlocked check). Add a spec that it is hidden on a locked project.
- Make updateSelectAllStatus null-safe so toggling a row checkbox on a locked
  project (where Select All is absent) no longer throws.
- Clean up the extracted updateBulkDeleteStatus indentation and file newline.
- De-flake the madmin projects spec via CGI.escapeHTML (same fix as PR #333).

JuanVqz committed:
Assert Bulk Delete stays disabled on a locked project when a story is selected

Documents that updateBulkDeleteStatus's unlocked guard is load-bearing: story
checkboxes still render on locked projects, so without the guard, selecting a
story would enable Bulk Delete.

JuanVqz committed:
Move Select All into a dedicated leftmost checkbox column

Give the checkboxes their own column with a bold 'Select All' header, instead
of tucking the control into the right-hand actions cell where it was hard to
find. Add the sixth grid track for the new column and shift the story-title
filter and estimate-position helper to account for it.
